Unit tests fail on Linux Mint Cinnamon 19.3

I believe that this is related to: https://github.com/emitter-io/emitter/issues/369

```bash
$ go version
go version go1.16.5 linux/amd64

$ go test ./...
resolving 
unable to resolve an empty name or url
resolving https://google.com
resolving google.com
resolving google.com/1234/sds
resolving https://127.0.0.1
resolving https://4987e9gs99sxdg8e8e7tgwe5.boomshouldnotwork
lookup 4987e9gs99sxdg8e8e7tgwe5.boomshouldnotwork: no such host
resolving 127.0.0.1
resolving 2001:db8::68
resolving 192.0.2.1
resolving 127.0.0.1:22
resolving 127.0.0.1:8xx0
parse "tcp://127.0.0.1:8xx0": invalid port ":8xx0" after host
resolving tcp://127.0.0.1:8000
resolving https://gooo:oogle.com
parse "https://gooo:oogle.com": invalid port ":oogle.com" after host
resolving https://google.com:xxx
parse "https://google.com:xxx": invalid port ":xxx" after host
resolving https-----:// ://--
parse "https-----:// ://--": invalid character " " in host name
resolving ::
resolving :::
resolving ::::::::
lookup :::::::: no such host
--- FAIL: TestResolve (0.16s)
    dns_test.go:48: 
        	Error Trace:	dns_test.go:48
        	Error:      	Not equal: 
        	            	expected: true
        	            	actual  : false
        	Test:       	TestResolve
address 2603:9001:8d01:f500:1198:e1ce:b639:9949:80: too many colons in address
unable to parse an empty address
address 2603:9001:8d01:f500:1198:e1ce:b639:9949:80: too many colons in address
address 2603:9001:8d01:f500:1198:e1ce:b639:9949:1234: too many colons in address
--- FAIL: TestParse (0.15s)
    ipaddr_test.go:31: 
        	Error Trace:	ipaddr_test.go:31
        	Error:      	Not equal: 
        	            	expected: false
        	            	actual  : true
        	Test:       	TestParse
    ipaddr_test.go:31: 
        	Error Trace:	ipaddr_test.go:31
        	Error:      	Not equal: 
        	            	expected: false
        	            	actual  : true
        	Test:       	TestParse
    ipaddr_test.go:31: 
        	Error Trace:	ipaddr_test.go:31
        	Error:      	Not equal: 
        	            	expected: false
        	            	actual  : true
        	Test:       	TestParse
FAIL
FAIL	github.com/emitter-io/address	0.314s
FAIL
```

--- a/dns.go
+++ b/dns.go
@@ -38,7 +38,7 @@ func Resolve(name string, defaultPort int) ([]net.TCPAddr, error) {
 	}
 
 	// The hostname might not be there (if it's an address)
-	if url.Hostname() != "" {
+	if url.Hostname() != "" && url.Hostname() != "::" {
 		name = url.Hostname()
 	}
 

--- a/go.mod
+++ b/go.mod
@@ -1,1 +1,5 @@
 module github.com/emitter-io/address
+
+go 1.16
+
+require github.com/stretchr/testify v1.7.0

--- a/ipaddr.go
+++ b/ipaddr.go
@@ -30,24 +30,40 @@ func Parse(addr string, defaultPort int) (*net.TCPAddr, error) {
 	// Convenience: set private address
 	if strings.Contains(addr, "private") {
 		private := GetPrivateOrDefault(Loopback)
-		addr = strings.Replace(addr, "private", private.String(), 1)
+		if private.IP.To4() == nil && strings.Contains(addr, ":") {
+			addr = strings.Replace(addr, "private", fmt.Sprintf("[%v]", private.String()), 1)
+		} else {
+			addr = strings.Replace(addr, "private", private.String(), 1)
+		}
 	}
 
 	// Convenience: set public address
 	if strings.Contains(addr, "external") {
 		external := GetExternalOrDefault(Loopback)
-		addr = strings.Replace(addr, "external", external.String(), 1)
+		if external.IP.To4() == nil && strings.Contains(addr, ":") {
+			addr = strings.Replace(addr, "external", fmt.Sprintf("[%v]", external.String()), 1)
+		} else {
+			addr = strings.Replace(addr, "external", external.String(), 1)
+		}
 	}
 
 	// Convenience: set public address
 	if strings.Contains(addr, "public") {
 		public := GetPublicOrDefault(Loopback)
-		addr = strings.Replace(addr, "public", public.String(), 1)
+		if public.IP.To4() == nil && strings.Contains(addr, ":") {
+			addr = strings.Replace(addr, "public", fmt.Sprintf("[%v]", public.String()), 1)
+		} else {
+			addr = strings.Replace(addr, "public", public.String(), 1)
+		}
 	}
 
 	// If we have only an IP address, use the default port
 	if ip := net.ParseIP(addr); ip != nil {
-		addr = fmt.Sprintf("%s:%d", ip, defaultPort)
+		if ip != nil && ip.To4() == nil {
+			addr = fmt.Sprintf("[%s]:%d", ip, defaultPort)
+		} else {
+			addr = fmt.Sprintf("%s:%d", ip, defaultPort)
+		}
 	}
 
 	// Resolve the address

